jetson/nav/rover.cpp

```diff
@@ -242,6 +261,47 @@ bool Rover::updateRover( RoverStatus newRoverStatus )
             mRoverStatus.odometry() = newRoverStatus.odometry();
             mRoverStatus.leftTarget() = newRoverStatus.leftTarget();
             mRoverStatus.rightTarget() = newRoverStatus.rightTarget();
+
+            // Perform basic caching, include special case when
+            // we store for the first time, but store an empty one
+            // so we don't just have "junk" in the target Cache.
+            if ( mRoverStatus.leftTarget().distance != -1 ) 
+            {
+                mRoverStatus.leftCacheTarget() = mRoverStatus.leftTarget();
+                mRoverStatus.getMisses() = 0;
+            }
+            else 
+            { // if this is empty, then so is rightTarget. increment misses
+                mRoverStatus.getMisses()++;
+            }
+
+            // Note: The only time this updates is ONLY if the leftTarget updates (by default
+            // the first target seen will be stored in leftTarget, and even if only 1 is seen
+            // this won't update, so we should be fine)
+            if ( mRoverStatus.rightTarget().distance != -1 ) 
+            {
+                mRoverStatus.rightCacheTarget() = mRoverStatus.rightTarget();
+                mRoverStatus.getMisses() = 0;
+            }
+
```

Comment:
you don't need to increment misses here in an else statement?

Reply:
I only have it increment on leftTarget misses, since the was the original detection was designed (and to keep the "queue" style of storing targets) was the first target detected is always going to be stored as the left one until another target is detected. Since if the leftTarget misses, that means so will the rightTarget. You bring up a point, if we see the leftTarget but don't see the rightTarget, that could be an issue.

Reply:
it should treat each target's cache as independent, if one is available and the other isn't you should use the cache for the one that isn't and the ground truth for the one that is.

Reply:
Remember that most of the time, we don't need to worry about the rightTarget, since that is only used for the gate (which there's only 1 of, as opposed to 4 in the past); and that mostly we;ll be needing to worry about Tagged posts. I'll consider changing it in the future since you bring up a good point